core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -377,6 +380,66 @@ public void addSupportedLanguage(Locale locale) {
     }
 
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public <T> void registerDependency(Class<? extends T> clazz, T instance){
+        registerDependency(clazz, clazz.getName(), instance);
+    }
+
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param key the key which needs to be present if that
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public <T> void registerDependency(Class<? extends T> clazz, String key, T instance){
+        if(dependencies.containsRow(clazz) && dependencies.containsColumn(key)){
+            throw new IllegalStateException("There is already an instance of " + clazz.getName() + " with the key " + key + " registered!");
+        }
+
+        dependencies.put(clazz, key, instance);
+    }
+
+    /**
+     * Attempts to inject instances of classes registered with {@link CommandManager#registerDependency(Class, Object)}
+     * into all fields that are marked with {@link Dependency}.
+     *
+     * @param baseCommand the instance which fields should be filled
+     */
+    void injectDependencies(BaseCommand baseCommand){
+        for (Field field : baseCommand.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(Dependency.class)) {
+                Dependency dependency = field.getAnnotation(Dependency.class);
+                String key = (key = dependency.value()).equals("") ? field.getType().getName() : key;
+                Object object = dependencies.row(field.getType()).get(key);
+                if(object == null){
+                    throw new UnresolvedDependencyException("Could not find a registered instance of " +
+                            field.getType().getName() + " with key " + key + " for field " + field.getName() +
+                            " in class " + baseCommand.getClass().getName());
+                }
+
+                try {
+                    field.setAccessible(true);
+                    field.set(baseCommand, object);
+                    field.setAccessible(false);
```

Comment:
Don't know if theres a reason to set this to false? what if it was already accessible before.